cartographer/mapping_2d/sparse_pose_graph/spa_cost_function.h

```diff
@@ -81,6 +81,56 @@ class SpaCostFunction {
   const Constraint::Pose pose_;
 };
 
+class WarpedSpaCostFunction {
```

Comment:
I suggest leaving this out for now. Let's keep the focus of this PR on introducing odometry to the optimization. Adding scaling factors can be done in a follow up PR.